src/main.cpp

```diff
@@ -7,10 +7,48 @@
 
 #include <QApplication>
 #include <QCommandLineParser>
+#include <QLocalServer>
+#include <QLocalSocket>
 #include <QLoggingCategory>
+#include <QMessageBox>
+
+void showErrorMessage(const char *text) {
+  QMessageBox msgBox;
+
+  msgBox.setIcon(QMessageBox::Critical);
+  msgBox.setText(text);
+  msgBox.setStandardButtons(QMessageBox::Ok);
+  msgBox.setDefaultButton(QMessageBox::Ok);
+  msgBox.setWindowIcon(QIcon(":assets/de.shorsh.discord-screenaudio.png"));
+
+  msgBox.exec();
+}
+
+bool isProgramRunning(const QString &program_name) {
+  QLocalSocket socket;
+  socket.connectToServer(program_name);
+  if (socket.waitForConnected()) {
+    return true; // program is already running
+  }
+  return false;
+}
 
 int main(int argc, char *argv[]) {
   QApplication app(argc, argv);
+
+  // Check if discord is already running
+  QString program_name = "discord-screenaudio";
+  if (isProgramRunning(program_name)) {
+    // if running show error message
+    showErrorMessage("Discord is already running");
```

Comment:
To avoid confusions with the official discord client I would maybe do

```suggestion
    showErrorMessage("discord-screenaudio is already running");
```

Reply:
Ok, i changed it to "discord-screenaudio"